src/krisi/evaluate/scorecard.py

```diff
@@ -1,29 +1,209 @@
+import logging
+from collections.abc import Iterable
 from dataclasses import dataclass
 from enum import Enum
-from typing import Any, Optional
+from typing import Any, Generic, List, Optional, Tuple, TypeVar, Union
+
+import numpy as np
+
+from krisi.utils.printing import bold, get_term_size, iterative_length
 
 
 class SampleTypes(Enum):
     insample = "insample"
     outsample = "outsample"
 
 
+class MCats(Enum):
+    residual = "Residual Diagnostics"
+    entropy = "Information Entropy"
+    class_err = "Forecast Errors - Classification"
+    reg_err = "Forecast Errors - Regression"
+
+
+T = TypeVar("T", bound=Union[float, int, str, List, Tuple])
+
+
+@dataclass
+class Metric(Generic[T]):
+    name: str
+    category: Optional[MCats] = None
+    metric_result: Optional[T] = None
+    hyperparameters: Optional[Any] = None
+    info: str = ""
+
+    def __setitem__(self, key: str, item: Any) -> None:
+        setattr(self, key, item)
+
+    def __getitem__(self, key: str) -> Any:
+        return getattr(self, key, "Unknown Field")
+
+    def __str__(self) -> str:
+        return print_metric(self)
+
+    def __repr__(self) -> str:
+        return print_metric(self, repr=True)
+
+
+def map_newdict_on_olddict(old_dict: dict, new_dict: dict) -> dict:
```

Comment:
there are many ways to merge dictionaries, like the `|` operator https://favtutor.com/blogs/merge-dictionaries-python